src/js/covidMap.js

```diff
@@ -761,40 +787,61 @@ fluid.covidMap.isPostcodeStart = function (query) {
     return query.match(/[a-z][0-9]/i);
 };
 
-fluid.covidMap.doQuery = function (rows, query, activeChecks, checks) {
+/** Perform the query, producing a set of matched rows
+ * @param {Object[]} rows - The data rows as loaded from CSV
+ * @param {String} query - The user's query string as present in the search box or imputed from the city selection
+ * @param {Boolean[]} activeChecks - The state of currently active accessibility filter checks
+ * @param {Object<String, Object>} filters - The filter check configuration structure, mapping keys to records encoding column
+ * @return {MatchElement[]} An array of match elements, containing booleans {match, matchQuery, matchFilters} indicating which part of the query state has matched
+ */
+fluid.covidMap.doQuery = function (rows, query, activeChecks, filters) {
     var normalised = query.trim();
     var isPostcodeQuery = fluid.covidMap.isPostcodeStart(normalised);
-    var noChecks = fluid.hashToArray(activeChecks).every(a => !a);
-    // var checksActive = activeChecks.some(a => a) && activeChecks.some(a => !a);
     var anyMatched = false;
     var matched = rows.map(function (row) {
         var matchQuery = (isPostcodeQuery ? row.postal_code : row.city).startsWith(query);
-        var checkMatches = fluid.transform(checks, function (record, key) {
+        var filterMatches = fluid.transform(filters, function (record, key) {
             var value = row[record.column];
-            return !activeChecks[key] || value.contains("Yes");
+            return !activeChecks[key] || value.includes("Yes");
         });
-        var matchChecks = noChecks || fluid.hashToArray(checkMatches).every(a => a);
-        var match = matchQuery && matchChecks;
+        var matchFilters = fluid.hashToArray(filterMatches).every(a => a);
+        var match = matchQuery && matchFilters;
         anyMatched = anyMatched || match;
-        return match;
+        return {
+            match: match,
+            matchQuery: matchQuery,
+            matchFilters: matchFilters // Note that this field is not currently read by anything
+        };
     });
     if (!anyMatched) {
```

Comment:
Does this condition block mean when there isn't any match found, show all testing centers and treat them as correct matches. I think it would be better to alert the user with "no match found". We probably need a design for this. What do you think?

Reply:
Sounds good to me

Reply:
As discussed at today's meeting, this will be addressed in a separate PR when the design for the "no match" is ready.